tests/system/test_apikey.py

```diff
@@ -0,0 +1,178 @@
+from apmserver import BaseTest, integration_test
+from elasticsearch import Elasticsearch
+import json
+import random
+
+
+class APIKeyBaseTest(BaseTest):
+    api_key_name = "apm_integration_key"
+
+    def config(self):
+        return {
+            "elasticsearch_host": self.get_elasticsearch_url(),
+            "file_enabled": "false",
+            "kibana_enabled": "false",
+        }
+
+    def setUp(self):
+        super(APIKeyBaseTest, self).setUp()
+        self.es = Elasticsearch([self.get_elasticsearch_url()])
+        self.kibana_url = self.get_kibana_url()
+        self.render_config_template(**self.config())
+
+    def subcommand_output(self, *args):
+        log = self.subcommand(*args)
+        # command and go test output is combined in log, pull out the command output
+        command_output = self._trim_golog(log)
+        return json.loads(command_output)
+
+    def subcommand(self, *args):
+        logfile = self.beat_name + "-" + str(random.randint(0, 99999)) + "-" + args[0] + ".log"
+        subcmd = ["apikey"]
+        subcmd.extend(args)
+        subcmd.append("--json")
+        self.run_beat(logging_args=[], extra_args=subcmd, output=logfile)
```

Comment:
You should pass the expected status code into `run_beat`, or return and check in the callers. It should be non-zero for the failing commands.